DQMServices/Core/interface/DQMEDAnalyzer.h

```diff
@@ -18,63 +22,42 @@ namespace dqmDetails {struct NoCache {};}
 
 
 class DQMEDAnalyzer
-    : public edm::stream::EDAnalyzer<edm::RunSummaryCache<dqmDetails::NoCache>,
-                                     edm::LuminosityBlockSummaryCache<dqmDetails::NoCache> >
+    : public edm::one::EDProducer<edm::one::WatchRuns,
+                                  edm::one::WatchLuminosityBlocks,
+                                  edm::Accumulator>
 {
 public:
   DQMEDAnalyzer();
   // implicit copy constructor
   // implicit assignment operator
   // implicit destructor
-  void beginStream(edm::StreamID id) final;
-  void beginRun(edm::Run const &, edm::EventSetup const&) final;
-  static std::shared_ptr<dqmDetails::NoCache> globalBeginRunSummary(edm::Run const&,
-                                                        edm::EventSetup const&,
-                                                        RunContext const*);
-  void endRunSummary(edm::Run const&,
+  virtual void beginJob() {};
+  virtual void beginRun(edm::Run const &, edm::EventSetup const&) final;
+  virtual void   endRun(edm::Run const &, edm::EventSetup const&) {};
+  virtual void analyze(const edm::Event&, const edm::EventSetup&) = 0;
+  virtual void accumulate(edm::Event const&, edm::EventSetup const&) override final;
+  virtual void endJob() {};
+  // TODO: Make sure those get called.
+  virtual void beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) {};
+  virtual void endLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) {};
+
+
+
+  /* (never called, only for stream-modules */ 
```

Comment:
why do you keep methods that you say are "never called" ?

Reply:
They should be removed, along with their implementation. However, I did not expect this migration to go as smoothly as it seems to be right now, and if we need a opt-out of some form, we still need them.

I'll either remove them later in this PR or in a follow-up.